piet/src/text.rs

```diff
@@ -94,6 +132,34 @@ pub trait TextLayout {
     fn hit_test_text_position(&self, text_position: usize) -> Option<HitTestTextPosition>;
 }
 
+/// Metadata about each line in a text layout.
+#[derive(Debug, Default)]
+pub struct LineMetric {
+    /// Line length (in code units), but offset from the beginning of the text. So it's the length
+    /// of this line summed with the lengths of all the lines before it.
+    ///
+    /// Does not include trailing whitespace.
+    pub line_length_offset: usize,
+
+    /// Line width, but offset from the beginning of the text. So it's the width
+    /// of this line summed with the widths of all the lines before it.
+    ///
+    /// Does not include trailing whitespace.
+    pub line_width_offset: f64,
+
+    /// Line length (in code units), but offset from the beginning of the text. So it's the length
+    /// of this line summed with the lengths of all the lines before it.
+    ///
+    /// Includes trailing whitespace.
+    pub line_length_trailing_whitespace_offset: usize,
```

Comment:
This particular one, if we're following DWrite, can maybe just be the size of the trailing whitespace. Otherwise, it should be the "trimmed line end" (ie line end - whitespace size).